service/lib/dinstaller/manager.rb

```diff
@@ -152,6 +142,21 @@ def install
 
   private
 
+    def initialize
+      Yast::Mode.SetUI("commandline")
+      Yast::Mode.SetMode("installation")
+      @languages = []
+      @products = []
+      @status_callbacks = []
+      @status = InstallerStatus::ERROR # it should start with probing, so just temporary status
+      @logger = logger || Logger.new($stdout)
```

Comment:
this looks wrong. logger will be always nil when initialized. We need to call instance.tap and set logger or maybe consider having singleton Logger object that we init very early and can modify it during runtime? What do you think?

Reply:
TBH, I haven't take care about the logger yet. Anyway, I haven't touched this code, ony moved it.